File: Easy/PHP/SortedSquaredArray.php
<?php

/**
 * Sort squared array.
 * O(nlog(n)) time | O(n) space
 *
 * @param array     $array     The input array.
 *
 * @return array
 */
function sortedSquaredArray1($array)
{
    $sorted = array();
    for ($x = 0; $x <= count($array) - 1; $x++) {
        $sorted[$x] = pow($array[$x], 2);
    }
    arsort($sorted);

    return $sorted;
}

/**
 * Sort squared array.
 * O(nlog(n)) time | O(n) space
 *
 * @param array     $array     The input array.
 *
 * @return array
 */
function sortedSquaredArray2($array)
{
    $start = 0;
    $end = count($array) - 1;
    $sorted = array();
    for ($x = count($array) - 1; $x >= 0; $x--) {
        if ($array[$start] < $array[$end]) {
            $sorted[$x] = pow($array[$end], 2);
            $end--;
        } else {
            $sorted[$x] = pow($array[$start], 2);
            $start++;
        }
    }

    return $sorted;
}

$result = sortedSquaredArray2([1, 2, 3, 5, 6, 8, 9]);
//var_dump($result);

print_r($result);

File: Easy/PYTHON/SortedSquaredArray.py
def sorted_squared_array1(array):
    sorted = [0 for _ in array]
    for i in range(0, len(array)):
        sorted[i] = pow(array[i], 2)

    sorted.sort()

    return sorted


def sorted_squared_array2(array):
    sorted = [0 for _ in array]
    start = 0
    end = len(array) - 1
    for i in reversed(range(len(array))):
        smallValue = array[start]
        largeValue = array[end]

        if abs(smallValue > largeValue):
            sorted[i] = smallValue * smallValue
            start += 1
        else:
            sorted[i] = largeValue * largeValue
            end -= 1

    return sorted


result = sorted_squared_array2([1, 5, 6, 8, 3])
print(result)
